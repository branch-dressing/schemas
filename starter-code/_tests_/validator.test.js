const {
    nameValidator,
    ageValidator,
    arrayValidator
} = require('../lib/Validator.js');

const dog = {
    name: 'Josh',
    age: 39,
    favoriteToys: ['bone', 'chew', 'ball']
};

const almostDog = {
    name: 'Nathan',
    age: '12',
    favoriteToys: () => {}
};

const notDog = {
    name: null,
    age: 'Twelve',
    favoriteToys: null
};

const emptyDog = {};

describe('validators', () => {
    it('can take an object and returns a fields value', () => {
        // Test that validate method can take an object and return a fields value
        expect(nameValidator.validate(dog)).toEqual('Josh');
        // required and field there and right type
        expect(nameValidator.validate(almostDog)).toEqual('Nathan');
        // required and field there but wrong type
        expect(() => nameValidator.validate(notDog)).toThrowErrorMatchingSnapshot();
        //required and field missing
        expect(() => nameValidator.validate(emptyDog)).toThrowErrorMatchingSnapshot();
        
        expect(ageValidator.validate(dog)).toEqual(39);
        expect(ageValidator.validate(almostDog)).toEqual(12);
        // Test that validate method can take an object and throw an error
        expect(() => ageValidator.validate(notDog)).toThrowErrorMatchingSnapshot();
        
        // not required and field there and right type
        expect(arrayValidator.validate(dog)).toEqual(['bone', 'chew', 'ball']);
        // not required and field there but wrong type
        expect(() => arrayValidator.validate(almostDog)).toThrowErrorMatchingSnapshot();
        // not required and field missing
        expect(arrayValidator.validate(notDog)).toEqual('Not Required');
    });
});
