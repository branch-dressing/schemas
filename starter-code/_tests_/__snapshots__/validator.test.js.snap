// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schemas when given an object it validates and casts that object 1`] = `"invalid schema >>Error: Cannot cast >>() => {}<< to Array"`;

exports[`Schemas when given an object it validates and casts that object 2`] = `"invalid schema >>Error: Cannot cast >>null<< to String,Error: Cannot cast >>Twelve<< to Number"`;

exports[`Schemas when given an object it validates and casts that object 3`] = `"invalid schema >>Error: Cannot cast >>undefined<< to String,Error: Cannot cast >>undefined<< to Number"`;

exports[`validators can take an object and returns a fields value 1`] = `"Cannot cast >>null<< to String"`;

exports[`validators can take an object and returns a fields value 2`] = `"Cannot cast >>undefined<< to String"`;

exports[`validators can take an object and returns a fields value 3`] = `"Cannot cast >>Twelve<< to Number"`;

exports[`validators can take an object and returns a fields value 4`] = `"Cannot cast >>() => {}<< to Array"`;
